### PR TITLE
Added support for AspNetCore

### PR DESCRIPTION
**Pull request by jspuij**
The real deal, asp.net core support, with Northwind sample.
Included is everything from the efcore branch as well, so either merge at once, or we do it in parts, and then I will need to rebase this once efcore is included.

Your DI changes were pretty easy to translate to aspnetcore, so kudos for that. Only thing I'm not sure about is the fact that we encapsulate the route building a bit now in aspnetcore, while that isn't needed in aspnetcore. We might want to look into that, but for now, this works.

Lets discuss!

### CODE REVIEW

**Comment by robertmclaws** [comment]
@jspuij I merged the EFCore PR, but there are some files now that are out of sync. You'll need to resolve before I can check this out.

Thanks!

**Comment by jspuij** [comment]
Done

**Comment by jspuij** [comment]
@robertmclaws I think I've adapted everything we discussed, but since our conversation was a few days ago and my notes weren't perfect, hopefully it's complete. Let me know.

**Review by robertmclaws — APPROVED** [review]
Looks great! Let's get it in and see how it goes.

File: src/Microsoft.Restier.Samples.Northwind.AspNetCore/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            services.AddRestier((builder) =>
+            {
+                // This delegate is executed after OData is added to the container.
+                // Add you replacement services here.
+                builder.AddRestierApi<NorthwindApi>(services =>
+                {
+
+                    services
+                        .AddDbContext<NorthwindContext>(opt => opt.UseSqlServer(Configuration.GetConnectionString("NorthwindEntities")))
```
[the remark this answers is not in the document]
Comment (jspuij) [reply src/Microsoft.Restier.Samples.Northwind.AspNetCore/Startup.cs]: I think you are right, but that's what I already did. I'm afraid Github is showing you the outdated code. I've got some issues with GitHub being slow to respond as well. It should be a single call, called `AddEFCoreProviderServices` with the overloaded arguments from `AddDbContext`.

**Comment by jspuij** [comment]
@robertmclaws  As always, thank you for taking the time to review this and to continue breathing life into this awesome overlooked library! @mikepizzo Good news, we got efcore and dotnet core support in here!

**Pull request merged**